utils/build-script

```diff
@@ -774,6 +778,115 @@ class BuildScriptInvocation(object):
 
         return options
 
+    def compute_product_classes(self):
```

Comment:
Do we really need to introduce a new concept called product classes. It seems like what you are really talking about are projects, no? Why not just make this a property called: "projects".